src/shims/io.rs

```diff
@@ -0,0 +1,213 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+
+use rustc::ty::layout::Size;
+
+use crate::stacked_borrows::Tag;
+use crate::*;
+
+pub struct FileHandle {
+    file: File,
+    flag: i32,
+}
+
+pub struct FileHandler {
+    handles: HashMap<i32, FileHandle>,
+    low: i32,
+}
+
+impl Default for FileHandler {
+    fn default() -> Self {
+        FileHandler {
+            handles: Default::default(),
```

Comment:
So what I do not understand here is, why is this needed? Why can't we just forward the raw FDs we got from the OS and let the program use them? What is achieved by adding this indirection?

Reply:
The main motivation for me at least was to be able to cross-interpret code, so this should work a windows host with a linux target.

Also it should be easy to extend this to work the other way around with windows' handles